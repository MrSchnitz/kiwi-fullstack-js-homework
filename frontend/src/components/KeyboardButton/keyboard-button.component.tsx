import React from "react";
import {Button} from "@kiwicom/orbit-components/lib";
import {Props} from "@kiwicom/orbit-components/lib/Button";
import {ButtonContent, ButtonLetters, ButtonNumber} from "./keyboard.styles";

export interface KeyboardButtonProps {
  number: number;
  onClick: (number: number) => void;
  charArray?: string[];
  props?: Props;
  isDelButton?: boolean;
}

const KeyboardButton: React.FC<KeyboardButtonProps> = ({
  number,
  charArray,
  onClick,
  props,
  isDelButton,
}: KeyboardButtonProps) => {
  return (
    <Button onClick={() => onClick(number)} {...props}>
      <ButtonContent>
        {!isDelButton && <ButtonNumber>{number}</ButtonNumber>}
        {charArray && (
          <ButtonLetters>
            {charArray.toString().replace(/,/g, "")}
          </ButtonLetters>
        )}
      </ButtonContent>
    </Button>
  );
};

export default KeyboardButton;
