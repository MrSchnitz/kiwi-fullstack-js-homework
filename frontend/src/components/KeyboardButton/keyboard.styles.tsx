import styled from "styled-components";

export const ButtonContent = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ButtonNumber = styled.h1`
  margin-right: 0.5rem;
`;

export const ButtonLetters = styled.h2``;
