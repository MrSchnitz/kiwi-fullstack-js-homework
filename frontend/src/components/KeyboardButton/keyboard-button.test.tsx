import React from "react";
import { shallow } from "enzyme";
import KeyboardButton, {
  KeyboardButtonProps,
} from "./keyboard-button.component";

describe("KeyboardButton component", () => {
  let wrapper: any;
  let mockOnClick: any;

  beforeEach(() => {
    mockOnClick = jest.fn();

    const mockProps: KeyboardButtonProps = {
      isDelButton: false,
      props: undefined,
      charArray: ["a", "b", "c"],
      number: 2,
      onClick: mockOnClick,
    };

    wrapper = shallow(<KeyboardButton {...mockProps} />);
  });

  it("should render KeyboardButton component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick when button is clicked", () => {
    wrapper.find("Button").simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
    expect(mockOnClick.mock.calls[0][0]).toBe(2);
  });
});
