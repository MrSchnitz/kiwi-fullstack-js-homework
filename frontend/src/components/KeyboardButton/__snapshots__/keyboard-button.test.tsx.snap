// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardButton component should render KeyboardButton component 1`] = `ShallowWrapper {}`;
